### ISSUE TITLE
Revert the conversion of list markers introduced in #83

### ISSUE THREAD

**Issue by jamesquilty** [executablebooks/mdformat#260]
### Description / Summary

This issue arises from discussion in #10 about the interpretation of certain Markdown list structures which are inherently ambiguous. _Originally posted by @hukkin in https://github.com/executablebooks/mdformat/issues/10#issuecomment-912705371_: 

> Consider the list
>
> ```markdown
> * item
> - item
> * item
> ```

This is a great example! The author's intended meaning is _inherently_ ambiguous. Did they mean a single list with three items or three lists with one item? There's no single correct interpretation based on the Markdown alone (although three lists with one item would be rather odd).

> If you render this as HTML you'll notice that there are three lists with one item in each.

No, different Markdown parsers produce different HTML depending on the interpretation of the authors. Case in point: while GitHub's parser produces three HTML lists, MacDown, which uses [hoedown](https://github.com/hoedown/hoedown) as its parser, produces a single HTML list:

```html
<ul>
<li>item</li>
<li>item</li>
<li>item</li>
</ul>
```

Even if the HTML differs, the rendered HTML remains ambiguous (see **Examples** below).

In the slightly less artificial example

```markdown
* item
* item

* item
* item
```

it's ambiguous whether the author intended a single list _and made an error_ or intended two lists. Again, different parsers will produce different HTML. GitHub produces the following HTML for the above Markdown source:

```html
<ul>
<li>
<p>item</p>
</li>
<li>
<p>item</p>
</li>
<li>
<p>item</p>
</li>
<li>
<p>item</p>
</li>
</ul>
```

which renders as

* item
* item

* item
* item

a single list with double-spaced items. A single list may be the author's intended meaning.

`mdformat` _assumes_ a particular intended meaning in these cases and, for the sake of preserving _assumed_ document semantics, makes changes which may _actually_ change the semantics of the document. For example, if/when `mdformat` changes the above Markdown to 

```markdown
* item
* item

- item
- item
```

Then GitHub's parser produces the following HTML:

```html
<ul>
<li>item</li>
<li>item</li>
</ul>
<ul>
<li>item</li>
<li>item</li>
</ul>
```

which renders as

* item
* item

- item
- item

While some ambiguity remains, the grouping implies _two_ lists with single-spaced items. This may not be the author's intended meaning.

Did `mdformat` preserve or did it change the document semantics?

We can never know the answer to this question because we do not know the author's intended meaning, nor whether they have made an error, nor the interpretation that Markdown parsers will make.

This is fundamentally an issue of _style_: it is _bad style_ for the author to write Markdown which is ambiguous and which produces ambiguous rendered HTML. The preferred solution is non-technical: authors avoid their problem of ambiguously expressing their meaning by always preceding consecutive lists with a descriptive sentence.

The technical solution implemented by #83 is based on an assumption of author intent and, in attempting to preserve _assumed_ document semantics can instead change the _actual_ document semantics.

#### Examples

What is the difference in meaning between

```html
<ul>
<li><p>item</p></li>
<li><p>item</p></li>
<li><p>item</p></li>
</ul>
```
and

```html
<ul>
<li>item</li>
</ul>
<ul>
<li>item</li>
</ul>
<ul>
<li>item</li>
</ul>
```

once they are rendered?

The above HTML produces

<ul>
<li><p>item</p></li>
<li><p>item</p></li>
<li><p>item</p></li>
</ul>

and

<ul>
<li>item</li>
</ul>
<ul>
<li>item</li>
</ul>
<ul>
<li>item</li>
</ul>

and it's ambiguous whether they are (a) the same single list with different item spacing _or_ (b) three lists with one item.

### Value / benefit

* #83 can create the very situation it sought to avoid, so it should be reverted.
* Reverting the changes introduced by #83 will simplify the code and lower maintenance workload.

### Implementation details

* This is a fairly simple reversion of #83

**Comment by chrisjsewell** [comment]
> No, different Markdown parsers produce different HTML depending on the interpretation of the authors.

Just to note here, that is not the case for CommonMark compliant parsers, which essentially is what mdformat supports.
Indeed there is an ongoing issue in hoedown where they want to support CommonMark: https://github.com/hoedown/hoedown/issues/119 (i.e. you should be pushing for that 😉)

The commonmark spec specifically has this stipulation: "Changing the bullet or ordered list delimiter starts a new list": https://spec.commonmark.org/0.30/#example-301

and this also applies to your comment in #10

> There's no single "correct" Markdown style, which is clearly the reason you've ended up writing multiple versions to "support" different styles: CommonMark, GFM and MyST.

again these all "inherit" from CommonMark, but then GFM and MyST add syntax extensions on top of it

**Comment by jamesquilty** [comment]
It seems that I misunderstood the source code and `mdformat` isn't doing the "if/when" I wrote above, so this issue should be closed.

@chrisjsewell thanks for taking the time to reply. We're talking cross-purposes here, I'm talking about _style_ and you're talking about _syntax_ and they're not the same thing. Ultimately, I've just spent less time and fewer characters (a total of 4 character changes) modifying `renderer/_util.py` and `_context.py` &mdash; effectively performing an  "opinionectomy" on `mdformat` so that it's compatible with the existing environment.
